### PR TITLE
Initial commit for lighthouse controller.

### PR DESCRIPTION
**Pull request by aswinsuryan**
Added a main file.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	<-runStoppedCh
+	klog.Info("All controllers stopped or exited. Stopping main loop")
+}
+
+func buildKubeFedFederator(stopCh <-chan struct{}) federate.Federator {
+	kubeConfig, err := clientcmd.BuildConfigFromFlags(masterURL, kubeConfig)
+	if err != nil {
+		klog.Fatalf("Error attempting to load kubeconfig: %s", err.Error())
+	}
+	federator, err := kubefed.New(kubeConfig)
```
[the remark this answers is not in the document]
Comment (aswinsuryan) [reply main.go]: while using latest I am getting the below error.

/pkg/mod/k8s.io/client-go@v11.0.0+incompatible/dynamic/interface.go:40:62: undefined: v1.PatchOptions

[the remark this answers is not in the document]
Comment (tpantelis) [reply main.go]: yeah I see that too. It seems there's a k8s version conflict with `github.com/coredns/coredns v1.5.2`. Looks like coredns uses a later version than submariner.

[the remark this answers is not in the document]
Comment (aswinsuryan) [reply main.go]: This commit is for creating a docker file and run lighthouse as a pod, I shall upgrade the version in my next commit where I will be using the watcher for service discovery. is that ok?

[the remark this answers is not in the document]
Comment (tpantelis) [reply main.go]: yeah that's fine.

**Comment by aswinsuryan** [comment]
This PR will not be required once https://github.com/submariner-io/lighthouse/pull/15 is merged. Hence closing.

**Pull request closed without merging**